Домашнее задание Модуль №16

Ulyanov-VN opened the pull request.

Ulyanov-VN committed:
Домашнее задание Модуль №16

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/player/domain/interactor/PlayerInteractor.kt:
```diff
@@ -30,7 +30,7 @@ class PlayerInteractorImpl(
         val previewUrl = track.previewUrl.orEmpty()
 
-        mediaRepository.initialize(previewUrl, object : com.example.playlistmaker.domain.repository.MediaCallbacks {
+        mediaRepository.initialize(previewUrl, object : com.example.playlistmaker.player.domain.repository.MediaCallbacks {
```
Тут путь до класса лучше заменить на корректный импорт

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/player/ui/PlayerActivity.kt:
```diff
@@ -0,0 +1,224 @@
+                    }
+                    is PlayerState.Stopped -> {
+                        updatePlayPauseButton(false)
+                        findViewById<TextView>(R.id.trackDuration).text = "00:00"
```
Лучше не хардкодить этот текст, так как форматирование может легко поменяться и потом придётся по всему проекту искать эти захардкоженные строчки. Можно либо вынести в strings.xml, чтобы это было только в одном месте, либо использовать ту же функцию форматирования через `SimpleDateFormat`, передавая туда значение 0, чтобы поведение было консистентным

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/player/ui/PlayerActivity.kt:
```diff
@@ -0,0 +1,224 @@
+                        findViewById<TextView>(R.id.trackDuration).text = "00:00"
+                    }
+                    is PlayerState.Prepared -> {
+                        findViewById<TextView>(R.id.durationValue).text = state.trackDuration
```
Вызовы `findViewById` являются дорогостоящими, так как на каждый вызов происходит поиск вью в иерархии, поэтому лучше результат этого вызова запоминать в переменной и переиспользовать

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/player/ui/viewmodels/PlayerState.kt:
```diff
@@ -0,0 +1,9 @@
+package com.example.playlistmaker.player.ui.viewmodels
+
+sealed class PlayerState {
```
Здесь больше подойдёт `sealed interface`, т.к. преимущества класса здесь не используются

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/sharing/domain/interactor/SharingInteractor.kt:
```diff
@@ -0,0 +1,20 @@
+}
+
+class SharingInteractorImpl : SharingInteractor {
+    override fun getShareAppContent(): String = "Check out this cool music app - Playlist Maker!"
```
Тут должна быть ссылка на курс практикума и все строчки должны лежать в ресурсах приложения - strings.xml

Ulyanov-VN committed:
Домашнее задание Модуль №16 (устранение замечаний)

1.❌ Ссылка на курс по Андроид-разработке находится в strings.xml и передаётся в Intent через метод контекста getString() - Ссылка на курс отсутствует, вместо ссылки захардкожен текст в SharingInteractorImpl
  ❌ Текст для каждого из предзаполненных полей электронного письма находятся в отдельных строках strings.xml и передаётся в Intent через метод контекста getString() - Все тексты захардкожены в SharingInteractorImpl
  ❌ Ссылка на оферту находится в strings.xml и передаётся в Intent через метод контекста getString() - Ссылка захардкожена в SharingInteractorImpl

  Исправлено.

2. ❌ Все взаимодействия со слоями Model должны осуществляться через ViewModel - SettingsActivity напрямую работает с SharingInteractor

  Исправлено.

Ulyanov-VN committed:
Домашнее задание Модуль №16 (устранение замечаний)

❌ Классы в пакете domain не используют классы из Android SDK - SharingInteractorImpl зависит от Context

Привет, исправил ошибку, класс SharingInteractorImpl больше не зависит от Android SDK (Context).
